src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.CacheEntryState.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    internal partial class CacheEntry
+    {
+        // this type exists just to reduce CacheEntry size by replacing many enum & boolean fields with one of a size of Int32
+        [StructLayout(LayoutKind.Explicit)]
```

Comment:
Why does this need to have explicit layout?

Reply:
The initial implementation had one in32 field that was overlapping four byte fields, but I've removed it in https://github.com/dotnet/runtime/pull/45962/commits/46ae2eadcaeed0e35166fe44302c1264413784bd and forgot to remove the attribute. I am going to do it right away